web_src/js/features/repo-settings-branches.ts

```diff
@@ -0,0 +1,30 @@
+import {createSortable} from '../modules/sortable.ts';
+import {POST} from '../modules/fetch.ts';
+
+export function initRepoBranchesSettings() {
+  const protectedBranchesList = document.querySelector(
+    '#protected-branches-list',
+  );
+  if (!protectedBranchesList) return;
+
+  createSortable(protectedBranchesList, {
+    handle: '.drag-handle',
+    animation: 150,
+    onEnd: async () => { // eslint-disable-line @typescript-eslint/no-misused-promises
```

Comment:
What is `misused` and why ignored?

Reply:
no idea

I just reused that code

https://github.com/go-gitea/gitea/blob/f2a6df03d953d608a5cac19cb9fa2c6d62dbe0e3/web_src/js/features/repo-issue-list.ts#L199

Reply:
PS: if I remove that the linter fails - but the code works ... so no idea whats going on

Reply:
The definition is in `export interface SortableOptions {`, `onEnd` doesn't accept `async`. 

The proper change is:

```
onMove: () => {
    (async () => {
         await ....
    })();
}
```

But I won't block it if you don't want to make further change.

Reply:
08df5593c263f9f7e4b64583bea36500080937d0